src/Hamiltonians/angular_momentum.jl

```diff
@@ -0,0 +1,125 @@
+"""
+    AxialAngularMomentumHO(S; z_dim = 3, addr = BoseFS(prod(S))) <: AbstractHamiltonian
+
+Angular momentum operator for application to Cartesian harmonic oscillator basis,
+see [`HOCartesianEnergyConserved`](@ref) or [`HOCartesianEnergyConservedPerDim`](@ref).
+Represents the projection of angular momentum onto `z`-axis:
+```math
+\\hat{L}_z = i \\hbar \\sum_{j=1}^N \\left( b_x b_y^\\dag - b_y b_x^\\dag \\right),
+```
+where ``b_x^\\dag`` and ``b_x`` are raising and lowering (ladder) operators 
+for  a harmonic oscillator in the ``x`` dimension, and simlarly for ``y``. 
+
+This is implemented for an ``N`` particle Fock space with creation and annihilation 
+operators as
+```math
+\\frac{1}{\\hbar} \\hat{L}_z = i \\sum_{n_x=1}^{M_x} \\sum_{n_y=1}^{M_y} 
+    \\left( a_{n_x-1,n_y+1}^\\dag - a_{n_x+1,n_y-1}^\\dag \\right) a_{n_x, n_y}.
+```
+in units of ``\\hbar``.
+
+Argument `S` is a tuple defining the range of Cartesian modes in each dimension and
+their mapping to Fock space modes in a `SingleComponentFockAddress`. If `S` indicates 
+a 3D system the `z` dimension can be changed by setting `z_dim`; 
+`S` should be be isotropic in the remaining `x`-`y` plane, i.e. must have 
+`S[x_dim] == S[y_dim]`.
+The starting address `addr` only needs to satisfy `num_modes(addr) == prod(S)`.
+
+# Example
+Calculate the overlap of two Fock addresses interpreted as harmonic oscillator states
+in a 2D Cartesian basis
+
+```jldoctest
+julia> S = (2,2)
+(2, 2)
+
+julia> Lz = AxialAngularMomentumHO(S)
+AxialAngularMomentumHO(BoseFS{0,4}((0, 0, 0, 0)), (2, 2); z_dim = 3)
+
+julia> v = DVec(BoseFS(prod(S), 2 => 1) => 1.0)
+DVec{BoseFS{1, 4, BitString{4, 1, UInt8}},Float64} with 1 entry, style = IsDeterministic{Float64}()
+  fs"|0 1 0 0⟩" => 1.0
+
+julia> w = DVec(BoseFS(prod(S), 3 => 1) => 1.0)
+DVec{BoseFS{1, 4, BitString{4, 1, UInt8}},Float64} with 1 entry, style = IsDeterministic{Float64}()
+  fs"|0 0 1 0⟩" => 1.0
+
+julia> dot(w, Lz, v)
+0.0 + 1.0im
+```
+"""
+struct AxialAngularMomentumHO{A,D} <: AbstractHamiltonian{ComplexF64}
+    addr::A
+    S::NTuple{D,Int64}
+    xyz::NTuple{3,Int64}
+end
+
+function AxialAngularMomentumHO(S; z_dim = 3, addr = BoseFS(prod(S)))
+    D = length(S)
+    D < 2 && throw(ArgumentError("number of dimensions should be at least 2"))
+    if D == 3 && z_dim ≠ 3
+        if z_dim == 1
+            x_dim, y_dim = (2,3)
+        elseif z_dim == 2
+            x_dim, y_dim = (1,3)
+        else
+            throw(ArgumentError("invalid choice of z dimension"))
+        end
+    else
+        x_dim, y_dim = (1, 2)
+    end
+    S[x_dim] == S[y_dim] || throw(ArgumentError("angular momentum only defined for isotropic system"))
+    prod(S) == num_modes(addr) || throw(ArgumentError("address type mismatch"))
+    return AxialAngularMomentumHO{typeof(addr),D}(addr, S, (x_dim, y_dim, z_dim))
+end
+
+function Base.show(io::IO, L::AxialAngularMomentumHO)
+    print(io, "AxialAngularMomentumHO($(L.addr), $(L.S); z_dim = $(L.xyz[3]))")
```

Comment:
```suggestion
    print(io, "AxialAngularMomentumHO($(L.S); z_dim = $(L.xyz[3]), addr = $(L.addr)")
```
With this change the output becomes valid input.

Note that this will require modifying the `jldoctest` example.

Reply:
Sorry, a closing parenthesis was missing:
```
print(io, "AxialAngularMomentumHO($(L.S); z_dim = $(L.xyz[3]), addr = $(L.addr))")
```